src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,34 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder builder = new StringBuilder("Report for period " + dateFrom + " - " + dateTo);
+        LocalDate localDateFrom = LocalDate.parse(dateFrom, FORMATTER);
+        LocalDate localDateTo = LocalDate.parse(dateTo, FORMATTER);
+        String infoSalary = null;
+        int[] salary = new int[names.length];
+
+        for (int i = 0; i < names.length; i++) {
+            for (int j = 0; j < data.length; j++) {
+                String[] splitData = data[j].split(" ");
+                LocalDate localSplitDate = LocalDate.parse(splitData[0], FORMATTER);
+                if ((localDateFrom.isBefore(localSplitDate)
+                        || localDateTo.isEqual(localSplitDate))
+                        && (localDateTo.isAfter(localSplitDate)
+                        || localDateTo.isEqual(localSplitDate))
+                        && names[i].equals(splitData[1])) {
+                    salary[i] = salary[i]
+                            + Integer.parseInt(splitData[2]) * Integer.parseInt(splitData[3]);
```

Comment:
Add constants for magic numbers (NAME_INDEX, ...)

Reply:
read about constant, pls
![image](https://user-images.githubusercontent.com/12864128/205956450-445166ed-cda6-46ec-9b35-6befe6a600ae.png)